types/index.d.ts

```diff
@@ -0,0 +1,153 @@
+import type { Client } from 'aedes';
+import type { AedesPacket } from 'aedes-packet';
+import type { Readable } from 'stream';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId?: string;
+  topic: string;
+  qos: number;
+}
+
+export interface Brokers {
+  [brokerId: string]: {
+    brokerId: string;
+  };
+}
```

Comment:
Also this one, brokers is an object `brokerId : lastHearthbeatTimestamp`, it's from `aedes.brokers`

Ref: https://github.com/moscajs/aedes/blob/master/aedes.js#L142
https://github.com/moscajs/aedes/blob/master/aedes.js#L103

Reply:
Right, thanks for giving me a clue. By the way that would be nice to add that type in Aedes.

Reply:
Absolutely. PR?

Reply:
Sure, later this week.
I noticed in `outgoingEnqueue` and `outgoingEnqueueCombi` that the clientId is expected in the subscriptions and since it's called by aedes, so i could also extend `mqtt-packet` Subscription with `{clientId: string}`, right ?

Reply:
yep !